src/core/deploy.ts

```diff
@@ -0,0 +1,104 @@
+import * as Config from '@oclif/config';
+import debug from 'debug';
+
+import { API } from '../definition';
+import { BumpApi } from '../api';
+import { VersionRequest, VersionResponse } from '../api/models';
+
+export class Deploy {
+  _bump!: BumpApi;
+  _config: Config.IConfig;
+
+  public constructor(config: Config.IConfig) {
+    this._config = config;
+  }
+
+  public async run(
+    file: string,
+    dryRun: boolean,
+    documentation: string,
+    token: string,
+    hub: string | undefined,
+    autoCreate: boolean,
+    documentationName: string | undefined,
+    branch: string | undefined,
+  ): Promise<VersionResponse | undefined> {
+    let version: VersionResponse | undefined = undefined;
+
+    const api = await API.load(file);
+    const [definition, references] = api.extractDefinition();
+
+    this.d(`${file} looks like an ${api.specName} spec version ${api.version}`);
+
+    const request: VersionRequest = {
+      documentation,
+      hub,
+      documentation_name: documentationName,
+      auto_create_documentation: autoCreate && !dryRun,
+      definition,
+      references,
+      branch_name: branch,
+    };
+
+    if (dryRun) {
+      await this.validateVersion(request, token);
+    } else {
+      version = await this.createVersion(request, token);
+    }
+
+    return version;
```

Comment:
What is the value of `response` when `dryRun` is true?

Reply:
The value of `version` is `undefined` (as defined at line 26: `let version: VersionResponse | undefined = undefined;`).